src/ORM/Marshaller.php

```diff
@@ -133,7 +138,7 @@ public function one(array $data, array $options = [])
             $columnType = $schema->columnType($key);
             if (isset($propertyMap[$key])) {
                 $assoc = $propertyMap[$key]['association'];
-                $value = $this->_marshalAssociation($assoc, $value, $propertyMap[$key]);
+                $value = $this->_marshalAssociation($assoc, $value, $propertyMap[$key] + $marshallOptions);
```

Comment:
Well, to be specific, this is the piece code we are talking about.

https://github.com/Tzaoh/cakephp/blob/32a1c60d795caa7b2a26fd13b7e1d002b6019e38/src/ORM/Marshaller.php#L239-L253

aparently if I do not specify a concrete key the flow will end in `$marshaller->many(...)`
So I think it will create new rows if the primaryKey(s) is specified.

Although I will check it as soon as I grab my pc.